src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -40,6 +41,7 @@
 tsql_dialect.sets("reserved_keywords").clear()
 tsql_dialect.sets("unreserved_keywords").clear()
 tsql_dialect.sets("reserved_keywords").update(RESERVED_KEYWORDS)
+tsql_dialect.sets("reserved_keywords").difference_update(BUILTIN_FUNCTIONS)
```

Comment:
Is there a reason you used `difference_update()` rather than just `update()`? This _seems_ like a potential bug. Note that `difference_update()` will **remove** items from the set.

https://www.programiz.com/python-programming/methods/set/difference_update

Reply:
mistake of mine. thanks.